### PR TITLE
[Catalog] Primary Demo now will be presented as a contained button.

### PR DESCRIPTION
**Pull request by mohammadcazig**
Pivotal: https://www.pivotaltracker.com/story/show/156998584
screen shot 2018-04-25 at 11 28 11 am

### CODE REVIEW

**Review by jverkoey — CHANGES_REQUESTED** [review]
Theme the button with the app theme please.

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 1]
```diff
@@ -50,11 +51,71 @@ class NodeViewTableViewDemoCell: UITableViewCell {
   }
 }
 
+class NodeViewTableViewPrimaryDemoCell: UITableViewCell {
+
+  let containedButton = MDCButton()
+
+  override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+    setupContaintButton()
```
Comment (jverkoey) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: Contained

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 2]
```diff
@@ -50,11 +51,71 @@ class NodeViewTableViewDemoCell: UITableViewCell {
...
+  }
+
+  required init(coder: NSCoder) {
+    super.init(coder: coder)!
+  }
```
Comment (jverkoey) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: This should call setupContainedButton as well

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 3]
```diff
@@ -50,11 +51,71 @@ class NodeViewTableViewDemoCell: UITableViewCell {
...
+
+  override func prepareForReuse() {
+    super.prepareForReuse()
+  }
```
Comment (jverkoey) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: Remove if not needed.

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 4]
```diff
@@ -50,11 +51,71 @@ class NodeViewTableViewDemoCell: UITableViewCell {
...
+
+  func setupContaintButton() {
+    containedButton.setTitle("Start Demo", for: .normal)
+    let buttonScheme = MDCButtonScheme()
```
Comment (jverkoey) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: Use the app's global theme (AppTheme.globalTheme).

**Review by jverkoey — APPROVED** [review]

**Review by jverkoey — DISMISSED** [review]
Oops - changes actually requested.

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 1 above)
Comment (mohammadcazig) [reply catalog/MDCCatalog/MDCNodeListViewController.swift]: Done

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 2 above)
Comment (mohammadcazig) [reply catalog/MDCCatalog/MDCNodeListViewController.swift]: Done.

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 3 above)
Comment (mohammadcazig) [reply catalog/MDCCatalog/MDCNodeListViewController.swift]: Done.

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 4 above)
Comment (mohammadcazig) [reply catalog/MDCCatalog/MDCNodeListViewController.swift]: Done.

**Comment by jverkoey** [comment]
Update the screenshot please.

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 5]
```diff
@@ -50,11 +51,68 @@ class NodeViewTableViewDemoCell: UITableViewCell {
...
+  }
+
+  required init(coder: NSCoder) {
+    super.init(coder: coder)!
+    setupContainedButton()
+  }
+
+  func setupContainedButton() {
+    containedButton.setTitle("Start Demo", for: .normal)
+    let buttonScheme = AppTheme.globalTheme.buttonScheme
```
Comment (jverkoey) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: You'll now need to register for `AppTheme.didChangeGlobalThemeNotificationName` notifications to ensure that this button's theme updates when the app theme changes.

**Review by jverkoey — DISMISSED** [review]

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 5 above)
Comment (jverkoey) [reply catalog/MDCCatalog/MDCNodeListViewController.swift]: Ah, this class already registers for the event. You just have to re-apply the theme in `themeDidChange`.

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 6]
```diff
@@ -319,34 +377,55 @@ extension MDCNodeListViewController {
...
-                                            reuseIdentifier: "NodeViewTableViewDemoCell")
-    }
-
-    var subtitleText: String?
+    var cell : UITableViewCell?
     if indexPath.section == Section.description.rawValue {
-      subtitleText = node.children[indexPath.row].exampleViewControllerName()
-      cell!.textLabel!.text = "Demo"
+      cell = tableView.dequeueReusableCell(withIdentifier: "NodeViewTableViewPrimaryDemoCell")
+      if cell == nil {
```
Comment (ianegordon) [inline catalog/MDCCatalog/MDCNodeListViewController.swift]: super-nit: You no longer need to test the results of deque against nil if you are using registered identifiers.
Here and below.

**Review by ianegordon — APPROVED** [review]
